balboa-http/src/main/scala/com/socrata/balboa/server/rest/EntitiesRest.scala

```diff
@@ -18,14 +18,18 @@ object EntitiesRest {
 
   def apply(params: Params): ResponseWithType = {
     val predicate: (String => Boolean) =
-      params.get("filter").map(Pattern.compile).map { pattern => (str: String) => pattern.matcher(str).matches }.getOrElse(_ => true)
+      params.get("filter").map(Pattern.compile)
+        .map { pattern => (str: String) => pattern.matcher(str).matches }.getOrElse(_ => true)
 
     val it = ds.entities().asScala.filter(predicate)
     // backwards-compatibility: -1 == no limit
     val limitString = params.getOrElse("limit", "-1")
     val limit = Try(limitString.toInt).recover({
       case err: NumberFormatException =>
-        return ResponseWithType(json, BadRequest(Error(400, "Unable to parse limit : " + JString(err.toString).toString.drop(1).dropRight(1))))
+        return ResponseWithType(json,
+          BadRequest(Error(
+            BadRequestCode,
+            "Unable to parse limit : " + JString(err.toString).toString.drop(1).dropRight(1))))
```

Comment:
The JString logic was there before I touched the codebase, and used to exist in several other places as well. I believe the sole purpose of it was to escape the String into a valid JSON String (escaping quotes, special characters, and the like). I wouldn't be surprised if that's unnecessary now I'm using json4s to automatically convert my case class to and from JSON. I'll remove it pending investigation.